### PR TITLE
Fix commands constructors and add unit tests

### PR DESCRIPTION
**Pull request by ylatuya**

### CODE REVIEW

File: VAS.Core/MVVMC/Command.cs [hunk 1]
```diff
@@ -219,21 +229,25 @@ public Command (Action<T> execute, Func<T, bool> canExecute) : base (o => execut
 	/// </summary>
 	public class AsyncCommand<T> : Command<T>
 	{
-		public AsyncCommand (Func<T, Task> execute) : base (o => execute (o))
+		public AsyncCommand (Func<T, Task> execute)
 		{
-			Contract.Requires (execute != null);
+			this.execute = o => execute ((T)o);
```
Comment (vguzmanp) [inline VAS.Core/MVVMC/Command.cs]: Shouldn't this keep the contract.Requires?

File: VAS.Tests/MVVMC/TestCommand.cs [hunk 2]
```diff
@@ -31,5 +31,70 @@ public void TestCommandInitExecutable ()
...
+
+			Assert.IsTrue (command.CanExecute (true));
+		}
+
+		[Test]
+		public void CanExecute_CommandParametrizedCanExecute_OK ()
+		{
+			var command = new Command<bool> (o => { }, o => o);
+
+			Assert.IsTrue (command.CanExecute (true));
```
Comment (vguzmanp) [inline VAS.Tests/MVVMC/TestCommand.cs]: Didn't we need to call `EmitCanExecute` or something like that to update the CanExecute? I mean, if you pass `false` here, as you haven't called the emit, will it return `false`?

Comment (ylatuya) [inline VAS.Tests/MVVMC/TestCommand.cs]: That's not how it works, If a funcion is passed in the constructor, CanExecute will always call that function , in this case o => o, or p => !projectsManager.Selection.Any() for example.
EmitCanExecute is just to notify views that the command execution state might have changed, so they should call again CanExecute to check the current state.

File: VAS.Core/MVVMC/Command.cs (hunk 1 above)
Comment (ylatuya) [inline VAS.Core/MVVMC/Command.cs]: yes, already fixed but not pushed

**Pull request merged**